state-chain/runtime/src/chainflip.rs

```diff
@@ -337,7 +337,13 @@ impl TransactionBuilder<Solana, SolanaApi<SolEnvironment>> for SolanaTransaction
 	fn build_transaction(
 		signed_call: &SolanaApi<SolEnvironment>,
 	) -> <Solana as Chain>::Transaction {
-		signed_call.transaction.clone()
+		SolanaTransactionData {
+			serialized_transaction: signed_call
+				.transaction
+				.clone()
+				.finalize_and_serialize()
+				.expect("Failed to serialize payload"),
```

Comment:
```suggestion
			serialized_transaction: signed_call.chain_encoded(),
```
this also removes the panic here

Reply:
`chain_encoded` will `unwrap_or_default`. I think if it fails to serialize we should not be defaulting, as that will just be an invalid transaction, no?

Reply:
Sure, an invalid transaction is much better than bricking the network

Reply:
Even if we don't want to unwrap_or_default, we should change the behaviour of `chain_encoded` in that case.